gooddata-sdk/gooddata_sdk/catalog/organization/service.py

```diff
@@ -60,3 +64,71 @@ def update_name(self, name: str) -> None:
         organization.attributes.name = name
         organization_document = CatalogOrganizationDocument(data=organization)
         self._entities_api.update_entity_organizations(organization.id, organization_document.to_api())
+
+    def create_or_update_jwk(self, jwk: CatalogJwk) -> None:
+        """Create a new jwk or overwrite an existing jwk with the same id.
+
+        Args:
+            jwk (CatalogJwk):
+                Catalog Jwk object to be created or updated.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError: Jwk can not be updated.
+        """
+        try:
+            self.get_jwk(jwk.id)
+            jwk_document = CatalogJwkDocument(data=jwk)
+            self._entities_api.update_entity_jwks(id=jwk.id, json_api_jwk_in_document=jwk_document.to_api())
+        except NotFoundException:
+            jwk_document = CatalogJwkDocument(data=jwk)
+            self._entities_api.create_entity_jwks(json_api_jwk_in_document=jwk_document.to_api())
+
+    def get_jwk(self, jwk_id: str) -> CatalogJwk:
+        """Get an individual jwk.
+
+        Args:
+            jwk_id (str):
+                Jwk identification string e.g. "demo"
+
+        Returns:
+            CatalogJwk:
+                Catalog jwk object containing structure of the jwk.
+        """
+        jwk_api = self._entities_api.get_entity_jwks(id=jwk_id).data
+        return CatalogJwk.from_api(jwk_api)
+
+    def delete_jwk(self, jwk_id: str) -> None:
+        """Delete a jwk.
+
+        Args:
+            jwk_id (str):
+                Jwk identification string e.g. "demo"
+
+        Returns:
+            None
+
+        Raises:
+            ValueError:
+                Jwk does not exist.
+        """
+        jwks = self.list_jwks()
+        if jwk_id not in [jwk.id for jwk in jwks]:
+            raise ValueError(f"Can not delete {jwk_id} jwk. This jwk does not exist.")
+        self._entities_api.delete_entity_jwks(jwk_id)
```

Comment:
IMHO, it is not necessary to list all jwks. I think we can wrap  `self._entities_api.delete_entity_jwks(jwk_id)` in try-except block and if the specific exception is raised, we can raise ValueError with the message.